src/components/AddBook.js

```diff
@@ -0,0 +1,196 @@
+import { useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import { nanoid } from 'nanoid'
+import placeholder from '../images/placeholder.png'
+import previewPlaceholder from '../images/preview-placeholder.png'
+
+function AddBook({ books, setBooks }) {
+  const [bookcover, setBookcover] = useState(placeholder)
+  const history = useHistory()
+  function getBookcoverPreview(previewEvent) {
+    const preview = URL.createObjectURL(previewEvent.target.files[0])
```

Comment:
Out of curiosity: What does this do exactly?

Reply:
I want to show a preview image after uploading the bookcover file in my addbook page and with URL.createObjectURL I can make a URL for this file and display it. :-) Read about it here: [URL.createObjectURL](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL)